src/components/Chart/data/dataAccumulator.ts

```diff
@@ -66,12 +70,276 @@ export interface DataAccumulator {
+                bitAccumulator.processBits(data.bits[i]);
+                bitAccumulator.processAccumulatedBits(timestamp);
+            }
+
+            ampereLineData[i] = {
+                x: timestamp,
+                y: v,
+            };
+        });
+
+        return {
+            ampereLineData,
+            bitsLineData: bitAccumulator.getLineData(),
+        };
+    }
+
+    const ampereLineData: AmpereState[] = new Array(noOfPointToRender * 2);
+
+    {
+        let min: number = Number.MAX_VALUE;
+        let max: number = -Number.MAX_VALUE;
+
+        let timestamp = begin + offset;
+        data.current.forEach((v, index) => {
+            const firstItemInGrp = index % numberOfPointsPerGrouped === 0;
+            const lastItemInGrp = (index + 1) % numberOfPointsPerGrouped === 0;
+            const grpIndex = Math.floor(index / numberOfPointsPerGrouped);
+
+            if (firstItemInGrp) {
+                min = Number.MAX_VALUE;
+                max = -Number.MAX_VALUE;
+            }
+
+            if (removeZeroValues && v === 0) {
+                v = NaN;
+            }
+
+            if (!Number.isNaN(v)) {
+                if (v > max) max = v;
+                if (v < min) min = v;
+
+                if (data.bits && index < data.bits.length) {
+                    bitAccumulator.processBits(data.bits[index]);
+                }
+            }
+
+            ampereLineData[grpIndex * 2] = {
+                x: timestamp,
+                y: min > max ? undefined : min,
+            };
+
+            ampereLineData[(grpIndex + 1) * 2 - 1] = {
+                x: timestamp,
+                y: min > max ? undefined : max,
+            };
+
+            if (lastItemInGrp) {
+                timestamp += timeGroup;
+                if (min <= max) {
+                    bitAccumulator.processAccumulatedBits(timestamp);
+                }
+            }
+        });
+    }
+
+    return {
+        ampereLineData,
+        bitsLineData: bitAccumulator.getLineData(),
+    };
+};
+
+const simplifyCurrentData = (
+    current: AmpereState[],
+    begin: number,
+    end: number
+) => current.filter(v => v.x !== undefined && v.x >= begin && v.x <= end);
+
+const simplifyDigitalChannelsData = (
+    dataChanel: DigitalChannelState[],
+    begin: number,
+    end: number
+) => {
+    if (dataChanel.length === 1) {
+        return [
+            {
+                x: begin,
+                y: dataChanel[0].y,
+            },
+            {
+                x: end,
+                y: dataChanel[0].y,
+            },
+        ];
+    }
+
+    if (dataChanel.length > 1) {
+        return [
+            {
+                x: begin,
+                y: dataChanel[0].y,
+            },
+            ...dataChanel.slice(1, dataChanel.length - 2),
+            {
+                x: end,
+                y: dataChanel[dataChanel.length - 1].y,
+            },
+        ];
+    }
+
+    return [];
+};
```

Comment:
This was an ugly workaround 

after further checking I believe that code was meant to be just 

``` if (dataChanel.length > 1) {
        return [
            {
                x: begin,
                y: dataChanel[0].y,
            },
            ...dataChanel.slice(1, dataChanel.length - 2),
            {
                x: end,
                y: dataChanel[dataChanel.length - 1].y,
            },
        ];
    }

    return [];
```

With this in mind:
Given my bitCachedData data come from the bit accumulation we only get the begin and end point and some data in between only if the bit state changes if not only two point will exist.

We need to simplify the cached data as we don't care of any data before our new begin. But doing a filter like `simplifyCurrentData` will leave us with possibly one point or non given that bitAccumulator only return the state change. Hence to simplify ur buffer we need to infer the value of at begin and at end and just update their timestamp to be what we decide as the new begin and end respectively.

also leaving the begin smaller that what we need to simplify the cache to will result in the chart not rendering correctly. In addition this will make joining very duffel in the case when the cache is missing information in front and/or back (e.g. when panning)